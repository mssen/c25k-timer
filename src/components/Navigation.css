.nav-button {
  cursor: pointer;
  border-radius: 4px;
  border: 0;
  color: white;
  font-size: 16px;
  background: gray;
  padding: 10px 20px 10px 20px;
  margin: 0px 5px;
}

.nav-button:hover {
  background: black;
}