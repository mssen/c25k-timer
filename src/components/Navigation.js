import React from 'react';

const Navigation = ({ week, day, triggerNavigation }) => {
  const handleNext = () => {
    triggerNavigation(1);
    return false;
  }

  const handlePrevious = () => {
    triggerNavigation(-1);
    return false;
  }

  return (
    <div>
      {
        !(week === 1 && day === 1)
          && <a href="#" onClick={ handlePrevious }>Week { day === 1 ? week - 1 : week }, Day { day === 1 ? 3 : day - 1 }</a>
      }
      ...
      {
        !(week === 9 && day === 3)
          && <a href="#" onClick={ handleNext }>Week { day === 3 ? week + 1 : week }, Day { day === 3 ? 1 : day + 1 }</a>
      }
    </div>
  )
}

export default Navigation;