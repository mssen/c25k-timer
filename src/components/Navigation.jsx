import React from 'react';
import PropTypes from 'prop-types';

import './Navigation.css';

const Navigation = ({ week, day, triggerNavigation, showModal }) => {
  const handleNext = () => {
    triggerNavigation(1);
    return false;
  };

  const handlePrevious = () => {
    triggerNavigation(-1);
    return false;
  };

  return (
    <div>
      {!(week === 1 && day === 1) && (
        <button className="nav-button" onClick={handlePrevious}>
          &lsaquo;&lsaquo; Week {day === 1 ? week - 1 : week}, Day {day === 1 ? 3 : day - 1}
        </button>
      )}
      <button className="nav-button" onClick={showModal}>
        ...
      </button>
      {!(week === 9 && day === 3) && (
        <button className="nav-button" onClick={handleNext}>
          Week {day === 3 ? week + 1 : week}, Day {day === 3 ? 1 : day + 1} &rsaquo;&rsaquo;
        </button>
      )}
    </div>
  );
};

Navigation.propTypes = {
  week: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  triggerNavigation: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired
};

export default Navigation;
