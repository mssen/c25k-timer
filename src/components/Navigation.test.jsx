import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

it('renders without crashing', () => {
  shallow(<Navigation week={ 1 } day={ 1 } triggerNavigation={ () => '' } showModal={ () => '' } />);
});

describe('renders the correct previous and next week/day', () => {
  it('on last day of week', () => {
    const wrapper = shallow(<Navigation week={ 1 } day={ 3 } triggerNavigation={ () => '' } showModal={ () => '' } />);
    expect(wrapper).toMatchSnapshot();
  });

  it('on first day of week', () => {
    const wrapper = shallow(<Navigation week={ 3 } day={ 1 } triggerNavigation={ () => '' } showModal={ () => '' } />);
    expect(wrapper).toMatchSnapshot();
  });

  it('in middle of week', () => {
    const wrapper = shallow(<Navigation week={ 1 } day={ 2 } triggerNavigation={ () => '' } showModal={ () => '' } />);
    expect(wrapper).toMatchSnapshot();
  });
});

it('handles navigation to the previous day', () => {
  const triggerNavigationStub = jest.fn();
  const wrapper = shallow(<Navigation week={ 1 } day={ 2 } triggerNavigation={ triggerNavigationStub } showModal={ () => '' } />);
  wrapper.find('button').at(0).simulate('click');
  expect(triggerNavigationStub).toBeCalledWith(-1);
});

it('handles navigation to the next day', () => {
  const triggerNavigationStub = jest.fn();
  const wrapper = shallow(<Navigation week={ 1 } day={ 2 } triggerNavigation={ triggerNavigationStub } showModal={ () => '' } />);
  wrapper.find('button').at(2).simulate('click');
  expect(triggerNavigationStub).toBeCalledWith(1);
});

it('handles show modal', () => {
  const showModalStub = jest.fn();
  const wrapper = shallow(<Navigation week={ 1 } day={ 2 } triggerNavigation={ () => '' } showModal={ showModalStub } />);
  wrapper.find('button').at(1).simulate('click');
  expect(showModalStub).toBeCalled();
});