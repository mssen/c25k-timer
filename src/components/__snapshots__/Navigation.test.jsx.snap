// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct previous and next week/day in middle of week 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    ‹‹ Week 
    1
    , Day 
    1
  </button>
  <button
    onClick={[Function]}
  >
    ...
  </button>
  <button
    onClick={[Function]}
  >
    Week 
    1
    , Day 
    3
     ››
  </button>
</div>
`;

exports[`renders the correct previous and next week/day on first day of week 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    ‹‹ Week 
    2
    , Day 
    3
  </button>
  <button
    onClick={[Function]}
  >
    ...
  </button>
  <button
    onClick={[Function]}
  >
    Week 
    3
    , Day 
    2
     ››
  </button>
</div>
`;

exports[`renders the correct previous and next week/day on last day of week 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    ‹‹ Week 
    1
    , Day 
    2
  </button>
  <button
    onClick={[Function]}
  >
    ...
  </button>
  <button
    onClick={[Function]}
  >
    Week 
    2
    , Day 
    1
     ››
  </button>
</div>
`;
